test/test_eventstream_rpc.py

```diff
@@ -92,9 +92,7 @@ def on_continuation_closed(self, **kwargs):
         self.record.close_call.set()
 
 
-@skipUnless(RUN_LOCALHOST_TESTS, "Skipping until we have permanent echo server")
-class TestClient(NativeResourceTest):
-
+class FailureClientTests(NativeResourceTest):
```

Comment:
Add comment about how these tests can run without a server
I had to think about why you split these up into 2 classes